flink-runtime/src/test/java/org/apache/flink/runtime/clusterframework/types/LocationPreferenceSlotSelectionStrategyTest.java

```diff
@@ -134,39 +139,57 @@ private Matcher<SlotSelectionStrategy.SlotInfoAndResources> withSlotInfo(SlotInf
                         resourceProfile, Collections.singletonList(nonLocalTm));
         Optional<SlotSelectionStrategy.SlotInfoAndLocality> match = runMatching(slotProfile);
 
-        Assert.assertTrue(match.isPresent());
-        final SlotSelectionStrategy.SlotInfoAndLocality slotInfoAndLocality = match.get();
-        assertThat(candidates, hasItem(withSlotInfo(slotInfoAndLocality.getSlotInfo())));
-        assertThat(slotInfoAndLocality, hasLocality(Locality.NON_LOCAL));
+        assertThat(match)
+                .hasValueSatisfying(
+                        slotInfoAndLocality -> {
+                            assertThat(slotInfoAndLocality.getLocality())
+                                    .isEqualTo(Locality.NON_LOCAL);
+                            assertThat(candidates)
+                                    .anySatisfy(
+                                            slotInfoAndResources ->
+                                                    assertThat(slotInfoAndResources.getSlotInfo())
+                                                            .isEqualTo(
+                                                                    slotInfoAndLocality
+                                                                            .getSlotInfo()));
+                        });
     }
 
     @Test
-    public void matchPreferredLocation() {
+    void matchPreferredLocation() {
 
         SlotProfile slotProfile =
                 SlotProfileTestingUtils.preferredLocality(
                         biggerResourceProfile, Collections.singletonList(tml2));
         Optional<SlotSelectionStrategy.SlotInfoAndLocality> match = runMatching(slotProfile);
 
-        Assert.assertEquals(slotInfo2, match.get().getSlotInfo());
+        assertThat(match)
+                .hasValueSatisfying(
+                        slotInfoAndLocality ->
+                                assertThat(slotInfoAndLocality.getSlotInfo()).isEqualTo(slotInfo2));
```

Comment:
The same logic appears in these two test class many times, I'd prefer extracts a specific method to handle this assertion.